### PR TITLE
chore: unify types for setTimeout return type to address the NodeJS types leak issue

### PR DESCRIPTION
**Pull request by peter-sanderson**
```ts
/**
 * Hack to workaround NodeJS.Timeout vs number type issue.
 *
 * Some reads on the topic:
 *  - https://stackoverflow.com/questions/45802988/typescript-use-correct-version-of-settimeout-node-vs-window
 *  - https://guilhermesimoes.github.io/blog/making-settimeout-return-number-in-typescript
 *
 */
 ```

### CODE REVIEW

File: packages/type-utils/src/timeout.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * Hack to workaround NodeJS.Timeout vs number type issue.
+ *
+ * Some reads on the topic:
+ *  - https://stackoverflow.com/questions/45802988/typescript-use-correct-version-of-settimeout-node-vs-window
+ *  - https://guilhermesimoes.github.io/blog/making-settimeout-return-number-in-typescript
+ *
+ */
+export type TimerId = ReturnType<typeof setTimeout>;
```
Comment (Nodonisko) [inline packages/type-utils/src/timeout.ts:9]: what about interval?

File: packages/coinjoin/src/client/CoinjoinPrison.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { TypedEmitter } from '@trezor/utils';
 
 import { CoinjoinPrisonInmate, CoinjoinPrisonEvents } from '../types/client';
 import { WabiSabiProtocolErrorCode } from '../enums';
+import { ImmediateId, TimerId } from '@trezor/type-utils/libDev/src';
```
Comment (Lemonexe) [inline packages/coinjoin/src/client/CoinjoinPrison.ts]: `libDev` import :scream:

File: packages/type-utils/src/timeout.ts (hunk 1 above)
Comment (Lemonexe) [reply packages/type-utils/src/timeout.ts:9]: indeed, we have some `ReturnType<typeof setInterval>` in our code, [e.g. here](https://github.com/trezor/trezor-suite/blob/b03e70cbc8d17e535e3d9ff5229596ee4b89e2e0/packages/blockchain-link/src/workers/electrum/client/caching.ts#L18)

**Comment by Lemonexe** [comment]
Looks good, but comments above, also lint errors..

File: packages/coinjoin/src/client/CoinjoinPrison.ts (hunk 2 above)
Comment (Nodonisko) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: this should be guarded by eslint, if not it was lost in latest refactor

**Review by Lemonexe — APPROVED** [review]
:rocket:

File: packages/coinjoin/src/client/CoinjoinPrison.ts (hunk 2 above)
Comment (Lemonexe) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: @Nodonisko hmm, I tried locally and eslint doesn't catch it, so FYI @peter-sanderson 
But you can resolve this thread :heavy_check_mark:

Comment (peter-sanderson) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: @Nodonisko do you recall how this was guarded? I am eager to bring it back. 

But I remember finding [this](https://github.com/trezor/trezor-suite/pull/14663) way before I merged the ESLint refactor.

Comment (Nodonisko) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: By this rule and wildcard https://eslint.org/docs/latest/rules/no-restricted-imports

Comment (peter-sanderson) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: Thanks I'll make sure to add it

Comment (peter-sanderson) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: I have found the config in past version of the suite https://github.com/trezor/trezor-suite/blob/release/24.4/.eslintrc.js#L222C10-L222C50

I will prepare test PR

Comment (peter-sanderson) [reply packages/coinjoin/src/client/CoinjoinPrison.ts]: Fix: https://github.com/trezor/trezor-suite/pull/15745

**Pull request merged**